### DISCUSSION TITLE
OTP login?

### DISCUSSION THREAD

**Discussion by benallfree** [pocketbase/pocketbase#5217] [Q&A] [answered]
Has anyone implemented an OTP login for PocketBase?

@ganigeorgiev Could I implement this by making an JS hook API endpoint that changes their pw to a 6 digit code and then emails it to them? then they input it on the frontend to log in?

Or is there some other way I can make the frontend client authenticate using OTP?

**Comment by ganigeorgiev** [comment]
OTP auth (and MFA) is already implemented as part of the [ongoing refactoring](https://github.com/pocketbase/pocketbase/discussions/4355) (there no ETAs yet).

You can implement it on your own in a similar manner to the gist in https://github.com/pocketbase/pocketbase/issues/119#issuecomment-1881906233.

**Reply by benallfree** [reply, marked as answer]
Thank you! I left my implementation here https://github.com/pocketbase/pocketbase/issues/119#issuecomment-2227944662
